presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoPageSource.java

```diff
@@ -181,7 +181,7 @@ private void writeSlice(BlockBuilder output, Type type, Object value)
     {
         String base = type.getTypeSignature().getBase();
         if (base.equals(StandardTypes.VARCHAR)) {
-            type.writeSlice(output, utf8Slice(value.toString()));
+            type.writeSlice(output, utf8Slice((value instanceof Document) ? ((Document) value).toJson() : value.toString()));
```

Comment:
Is `((Document) value).toJson()` consistent with how array of maps is printed?

I haven't found any test for the latter (probably should be added too).

Reply:
It is not the same as an array of map is printed. When a map is printed, it is not printed as JSON. If it is desired, I could change the way how an array of Document is printed as well.

Reply:
Let's keep it consistent, so either print both as JSON or as strings. Please add tests.

Reply:
Array of maps also follow the same convention.
Test for array of maps added as well.